src/signalr-hub.ts

```diff
@@ -47,7 +48,7 @@ export class SignalRHub {
     start() {
         if(!this.hasSubscriptions())
             console.warn('No listeners have been setup. You need to setup a listener before starting the connection or you will not receive data.');
-        this.connection.start();
+        this._primePromise = this.connection.start();
```

Comment:
change to something like this:

```
start() {
  this._conection.start()
      .then(() => this._prime$.next())
      .catch(err => this._prime$.error(err));
  return this._prime$.asObservable();
}
```

Reply:
See comment on using a `Subject` instead of a prime promise.